### ISSUE TITLE
Encapsulation of web front end vuetify

### ISSUE THREAD

**Issue by laotian2019** [sitewhere/sitewhere#850]
The sitewhere front-end style is built with vuetify. The author put vuetify.js It was encapsulated again. 
Excuse me? vuetify.js How is the encapsulation process written? 
How to use encapsulated vuetify.js Component. Or when adding styles, how can they not be encapsulated vuetify.js influence? Hope you can help me, thank you very much!

for example：
<sw-detail-page></sw-detail-page>    <sw-data-table-tab></sw-data-table-tab>

**Comment by derekadams** [comment]
We use our own IDE component library (https://github.com/sitewhere/sitewhere-ide-components) to provide the basis for consistent presentation across our user interface products. We don't advise making local updates to the SiteWhere admin UI since it's constantly evolving. We will be supporting plugins in the admin UI soon, but those would be built outside of the core and included dynamically. We do plan to release a web-oriented component library that supports some of the same functionality plus dashboards and graphs/charts soon, but it's independent of the admin user interface.

**Comment by yuyewuye** [comment]
We plan to add some of our own elements to the UI  and it's very limited if you don't have extensibility.During our testing ,We found some buttons such as delete, add, filter did not work well,therefore, we need to modify it ourselves.
Finally, can you provide documentation for some IDE component libraries?When will the new web-oriented component library released ? In 3.0 version ,will UI made change ?

**Comment by derekadams** [comment]
Yes, much of the 3.0 admin UI has been rewritten to offer a more consistent IDE-like experience. The libraries used are:
* **SiteWhere REST API** - https://www.npmjs.com/package/sitewhere-rest-api
* **SiteWhere IDE Components** - https://www.npmjs.com/package/sitewhere-ide-components
* **SiteWhere Admin UI Plugins** - https://www.npmjs.com/package/sitewhere-admin-ui-plugins
* **SiteWhere Admin UI** - https://github.com/sitewhere/sitewhere-admin-ui/tree/v3.0.0

Note that there have been significant changes for 3.0, but we have not released the final version yet. You can build the UI from the `v3.0.0` branch in order to see the latest progress. Most of the bugs have been fixed and we are actively adding new features now.

**Comment by yuyewuye** [comment]
It‘s very good . Would it be convenient if I wanted to expand? Because i want to some APP in Electron ,for example edge of computing or 3D visualization.

**Comment by derekadams** [comment]
I would advise developing applications outside of the admin UI as standalone applications since the admin application is targeted at system administrators and not end users. We are going to be working on a library of components which will allow you to build SiteWhere-based applications for the web and Electron more easily. This should enable what you are looking to build. Note that SiteWhere 3.0 also makes a core microservice library available so you can add your own microservices to the instance and manage them via the admin UI. We'll have documentation for that available soon.
